MHIdentityKit/Infrastructure/RequestAuthorizer/RequestAuthorizer.swift

```diff
@@ -40,6 +60,12 @@ extension URLRequest {
         authorizer.authorize(request: self, handler: handler)
     }
     
+    @available(iOS 13, tvOS 13.0.0, macOS 10.15, *)
+    public func authorize(using authorizer: RequestAuthorizer) async throws -> URLRequest {
```

Comment:
```suggestion
    public func authorized(using authorizer: RequestAuthorizer) async throws -> URLRequest {
```

In addition to that:
- mark the following functions with `@available(*, noasync)`, because they are using locks which should not be used from async context:
  - `public func authorized(using authorizer: RequestAuthorizer) throws -> URLRequest`
  - `public mutating func authorize(using authorizer: RequestAuthorizer) throws`
- create new async overload for the mutating authorize case, eg.
```
  @available(iOS 13, tvOS 13.0.0, macOS 10.15, *)
  public mutating func authorize(using authorizer: RequestAuthorizer) async throws {
    self = try await authorized(using: authorizer)
  }
```